Source/RapyutaSimulationPlugins/Public/Core/RRCoreUtils.h

```diff
@@ -0,0 +1,381 @@
+    static constexpr const TCHAR* CMD_RHI_ENABLE_GPU_CAPTURE_OPTIONS = TEXT("r.RHISetGPUCaptureOptions 1");
+    static constexpr const TCHAR* CMD_RENDER_DOC_CAPTURE_FRAME = TEXT("renderdoc.CaptureFrame");
+    static constexpr const TCHAR* CMD_SHADOW_MAP_CACHING_TURN_OFF = TEXT("r.Shadow.CacheWholeSceneShadows 0");
+    static constexpr const TCHAR* CMD_AO_USE_HISTORY_DISABLED = TEXT("r.AOUseHistory 0");
+    static constexpr const TCHAR* CMD_VISUALIZE_OCCLUDED_PRIMITIVES = TEXT("r.VisualizeOccludedPrimitives 1");
+
+    // SIM FILE EXTENSIONS --
+    static const TMap<ERRFileType, const TCHAR*> SimFileExts;
+    static FString GetFileTypeFilter(const ERRFileType InFileType);
+
+    FORCEINLINE static ERRFileType GetFileType(const FString& InFilePath)
+    {
+        ERRFileType imageFileType = ERRFileType::NONE;
+
+        for (uint8 i = 0; i < static_cast<uint8>(ERRFileType::TOTAL); ++i)
+        {
+            const ERRFileType& fileType = static_cast<ERRFileType>(i);
+            if (InFilePath.EndsWith(URRCoreUtils::SimFileExts[fileType]))
+            {
+                imageFileType = fileType;
+            }
+        }
+        return imageFileType;
+    }
+
+    FORCEINLINE static bool IsFileType(const FString& InFilePath, const TArray<ERRFileType>& InFileTypes)
+    {
+        for (const auto& fileType : InFileTypes)
+        {
+            if (InFilePath.EndsWith(URRCoreUtils::SimFileExts[fileType]))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+// SIM COMMAND LINE EXECUTION --
+#define CCMDLINE_ARG_FORMAT (TEXT("%s="))
+    static constexpr const TCHAR* CCMDLINE_ARG_INT_PHYSX_DISPATCHER_NUM = TEXT("physxDispatcher");
+    static void ExecuteConsoleCommand(const UObject* InContextObject, const FString& InCommandText)
+    {
+        GetPlayerController<APlayerController>(0, InContextObject)->ConsoleCommand(InCommandText);
+        // OR either one of these:
+        // + GEngine->Exec(World, *InCommandText);
+        // + World->Exec(World, *InCommandText);
+    }
+
+    // T must be primitive type only!
+    template<typename T>
+    static FORCEINLINE constexpr bool GetCommandLineArgumentValue(const TCHAR* InArgName, T& OutArgValue, bool bIsLogged = false)
+    {
+        bool bResult = false;
+        if constexpr (TIsSame<T, bool>::Value)
+        {
+            bResult = FParse::Bool(FCommandLine::Get(), *FString::Printf(CCMDLINE_ARG_FORMAT, InArgName), OutArgValue);
+        }
+        else
+        {
+            bResult = FParse::Value(FCommandLine::Get(), *FString::Printf(CCMDLINE_ARG_FORMAT, InArgName), OutArgValue);
+        }
+
+        if (!bResult && bIsLogged)
+        {
+            UE_LOG(LogTemp, Error, TEXT("Command line argument not found under the name [%s]"), InArgName);
+        }
+        return bResult;
+    }
+
+    template<typename T>
+    static FORCEINLINE constexpr bool ParseParams(const FString& InParams,

```

Comment:
where this function used?
if this is or  commandline execution,   it maybe better to rename with commandline~, since ParseParam is too generic name

Reply:
Right, it is not used yet, and will be used for running a custom automation test commandlet, or typically when users want to pass custom parameter to our Sim. Yeap, `ParseCommandlineParams` sounds better.